src/object/srv_obj.c

```
@@ -1592,6 +1592,13 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc, daos_iod_t *io
 	if (orw->orw_flags & ORF_CHECK_EXISTENCE)
 		goto out;
 
+	if (DAOS_FAIL_CHECK(DAOS_CLIENT_UNREACHABLE) && obj_rpc_is_update(rpc)) {
+		/** Fault injection - client unreachable. */
+		D_INFO("enabled fault injection client unreachable");
+		rc = -DER_CLIENT_UNREACH;
+		goto out;
+	}
```

is this PR just for fault injection?
I have missed where we can return that error code anywhere else from the server..

---

Yes, there are other PRs in flight but the point was to get the client code working in parallel with those other changes.

---

maybe a good candidate for a feature branch then?

---

If we move to a feature branch, are you ok with moving to a feature branch after this PR is landed?

---

Resolving this, please reopen as needed.

---

I hate to use a feature branch for a small feature.   This code is the majority of the feature.  The rest is just mercury/libfabtic and this can be tested independently.

---

well TBH this PR sounds a little less than half baked to land to master. Are the mercury and ofi changes landed?
i can't even tell how this is going to be used on the server side. you don't have to use a feature branch, just use this branch as a baseline..

---

We can continue to use this branch as a baseline. For more features built on this, we can make new child PRs that are based on this branch as a baseline and then slowly merge onto this PR.

---

the main advantage of a feature branch is that you can land PRs to it incrementally (have to be reviewed and passing tests same way as it it would be for master). then once all PRs in, the branch can be just landed to master without needing to review all the changes again. provided it passes CI testing of course.

---

We made a feature branch and changed this PR to merge my dev branch into this feature branch.

---

When you get a chance, @mchaarawi, can you please take a look again?

---

@mchaarawi @jolivier23 , sorry, it looks like for some reason Jenkins is not able to build my PR. Jeff told me the way around this is making a new PR. Please hold off on re-reviewing this while I get a new PR ready.